luarules/gadgets/game_unit_sharing_mode.lua

```diff
@@ -50,31 +50,37 @@ end
 -- Handles the specific condition for allowing /take transfers
 -- Returns true if the transfer should be allowed due to being a /take, false otherwise.
 local function CheckTakeCondition(fromTeamID, toTeamID)
-	-- Check if sender is allied and has no active human players
+	-- Check if sender is allied
 	if Spring.AreTeamsAllied(fromTeamID, toTeamID) then
-		local senderHasActivePlayer = false
+		-- Loop to see if sender has any active human players
 		for _, playerID in ipairs(Spring.GetPlayerList()) do
 			local _, active, spectator, teamID = Spring.GetPlayerInfo(playerID)
 			if active and not spectator and teamID == fromTeamID then
-				senderHasActivePlayer = true
-				break
+				-- Found an active player, so this is NOT the /take condition.
+				-- AllowUnitTransfer should proceed to check sharing rules.
+				return false
 			end
 		end
-		if not senderHasActivePlayer then
-			-- This matches the condition where the engine's /take command would trigger GiveEverythingTo
-			return true
-		end
+		-- If loop finished without finding an active player, it matches the /take condition.
+		-- Allow the transfer, bypassing sharing rules.
+		return true
 	end
-	return false -- Not a /take condition
+	-- Teams are not allied, not a /take condition.
+	return false
 end
 
+--[[ Determines if *this gadget* allows a unit transfer based on current rules.
+     Engine Behavior: The engine blocks the transfer if *any* gadget returns false.
+     Return Value: Returning true here only means *this* gadget doesn't object;
+       other gadgets might still block the transfer. Returning false blocks immediately.
+]]
 function gadget:AllowUnitTransfer(unitID, unitDefID, fromTeamID, toTeamID, capture)
 	-- 1. Check for clean override flags (e.g., Market)
 	for _, flagName in ipairs(transferOverrideFlags) do
 		local flagValue = Spring.GetUnitRulesParam(unitID, flagName)
 		if flagValue and flagValue == 1 then
 			-- Clear the flag after checking
-			Spring.SetUnitRulesParam(unitID, flagName, 0, {private=true})
+			Spring.SetUnitRulesParam(unitID, flagName, nil, {private=true})
```

Comment:
You can remove `, {private=true}` both here and in `unit_marked`. This does nothing.

Reply:
why is my robot lying to me? I will take your word for it.

Reply:
Is there a doc saying `private` is a thing? Can you link me to it?

This seems to have been deprecated at some point and does nothing on my investigation on the engine code